File: app/src/main/java/com/example/khughes/machewidget/db/OTAInfoDao.java
package com.example.khughes.machewidget.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.example.khughes.machewidget.OTAInfo;
import com.example.khughes.machewidget.OTAStatus.OTAStatus;
import com.example.khughes.machewidget.VehicleIds;
import com.example.khughes.machewidget.VehicleInfo;

import java.util.List;

@Dao
public interface OTAInfoDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertOTAInfo(OTAInfo info);

    @Query("SELECT * FROM ota_info WHERE ota_oemCorrelationId LIKE :correlationId AND vin LIKE :VIN "
            + "ORDER BY ota_dateTimestamp ASC" )
    List<OTAInfo> findOTAInfoByCorrelationId(String correlationId, String VIN);

    @Update
    void updateOTAInfo(OTAInfo info);
}

File: app/src/main/java/com/example/khughes/machewidget/db/OTAInfoDao.kt
package com.example.khughes.machewidget.db

import androidx.room.*
import com.example.khughes.machewidget.OTAInfo

@Dao
interface OTAInfoDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertOTAInfo(info: OTAInfo?)

    @Query(
        "SELECT * FROM ota_info WHERE ota_oemCorrelationId LIKE :correlationId AND vin LIKE :VIN "
                + "ORDER BY ota_dateTimestamp ASC"
    )
    fun findOTAInfoByCorrelationId(correlationId: String?, VIN: String?): List<OTAInfo?>?

    @Update
    fun updateOTAInfo(info: OTAInfo?)
}